### PR TITLE
Enable nullability in some DataGridViewCell members

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in some DataGridViewCell members.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10187)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewCell.cs [hunk 1]
```diff
@@ -1165,6 +1165,7 @@ private bool AccessibleRestructuringNeeded
         get
         {
             // Get the type of the editing control and cache it to not call expensive GetType() method repeatedly.
+            Debug.Assert(DataGridView?.EditingControl is not null);
```
[the remark this answers is not in the document]
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewCell.cs:1168]: Using `!` is my preference as well, but recently I was getting suggestions to use Debug.Assert. Check https://github.com/dotnet/winforms/pull/9704#discussion_r1350728551 for example.

**Comment by gpetrou** [comment]
@lonitra any thoughts on this one?

**Review by lonitra — APPROVED** [review]
LGTM

**Pull request merged**